src/httprpc.cpp

```diff
@@ -230,15 +235,17 @@ static bool HTTPReq_JSONRPC(const std::any& context, HTTPRequest* req)
             UniValue ret(UniValue::VARR);
             for (unsigned int reqIdx = 0; reqIdx < valRequest.size(); reqIdx++) {
                 // Batches never throw HTTP errors, they are always just included
-                // in "HTTP OK" responses.
+                // in "HTTP OK" responses. Notifications never get any response.
+                UniValue response;
                 try {
                     jreq.parse(valRequest[reqIdx]);
-                    ret.push_back(JSONRPCExec(jreq));
+                    response = JSONRPCExec(jreq);
                 } catch (const UniValue& objError) {
-                    ret.push_back(JSONRPCReplyObj(NullUniValue, objError, jreq.id, jreq.m_json_version));
+                    response = JSONRPCReplyObj(NullUniValue, objError, jreq.id, jreq.m_json_version);
                 } catch (const std::exception& e) {
-                    ret.push_back(JSONRPCReplyObj(NullUniValue, JSONRPCError(RPC_PARSE_ERROR, e.what()), jreq.id, jreq.m_json_version));
+                    response = JSONRPCReplyObj(NullUniValue, JSONRPCError(RPC_PARSE_ERROR, e.what()), jreq.id, jreq.m_json_version);
                 }
+                if (!jreq.IsNotification()) ret.push_back(response);
```

Comment:
In commit "rpc: JSON-RPC 2.0 should not respond to "notifications"" (bc2d55898b315d1ee80a70377d65f79c829246ed)

Would be good to `std::move(response)` here to avoid a copy